pkg/karpenter/karpenter_test.go

```diff
@@ -0,0 +1,77 @@
+package karpenter
+
+import (
+	"context"
+	"errors"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/weaveworks/eksctl/pkg/karpenter/providers/fakes"
+)
+
+var _ = Describe("InstallKarpenter", func() {
+	Context("Install", func() {
+		var (
+			fakeHelmInstaller *fakes.FakeHelmInstaller
+		)
+		BeforeEach(func() {
+			fakeHelmInstaller = &fakes.FakeHelmInstaller{}
+		})
+		AfterEach(func() {})
+		It("installs karpenter into an existing cluster", func() {
+			fakeHelmInstaller.AddRepoReturns(nil)
+			fakeHelmInstaller.InstallChartReturns(nil)
+			ki := &Installer{
+				Options: Options{
+					HelmInstaller:         fakeHelmInstaller,
+					Namespace:             "karpenter",
+					ClusterName:           "test-cluster",
+					AddDefaultProvisioner: true,
+					CreateServiceAccount:  true,
+					ClusterEndpoint:       "https://endpoint.com",
+					Version:               "0.4.3",
+				},
+			}
+			err := ki.InstallKarpenter(context.Background())
+			Expect(err).NotTo(HaveOccurred())
+		})
+		When("add repo fails", func() {
+			It("returns an error", func() {
+				fakeHelmInstaller.AddRepoReturns(errors.New("nope"))
+				ki := &Installer{
+					Options: Options{
+						HelmInstaller:         fakeHelmInstaller,
+						Namespace:             "karpenter",
+						ClusterName:           "test-cluster",
+						AddDefaultProvisioner: true,
+						CreateServiceAccount:  true,
+						ClusterEndpoint:       "https://endpoint.com",
+						Version:               "0.4.3",
+					},
+				}
+				err := ki.InstallKarpenter(context.Background())
+				Expect(err).To(MatchError(ContainSubstring("failed to karpenter repo: nope")))
+			})
+		})
```

Comment:
```suggestion
		})
	
```
you get the idea :smile:

Reply:
![tony](https://user-images.githubusercontent.com/182850/142876476-c0e64638-e544-4f95-9947-03077cee07e5.jpeg)